adapter/pkg/operator/apis/dp/v1alpha1/api_webhook.go

```diff
@@ -110,18 +110,26 @@ func (r *API) validateAPI() error {
 		r.Spec.APIType = "REST"
 	}
 
-	if !(r.Spec.ProdHTTPRouteRefs != nil && len(r.Spec.ProdHTTPRouteRefs) > 0) && !(r.Spec.SandHTTPRouteRefs != nil && len(r.Spec.SandHTTPRouteRefs) > 0) {
+	if !(r.Spec.Production != nil && r.Spec.Production[0].HTTPRouteRefs != nil && len(r.Spec.Production[0].HTTPRouteRefs) > 0) && !(r.Spec.Sandbox != nil && r.Spec.Sandbox[0].HTTPRouteRefs != nil && len(r.Spec.Sandbox[0].HTTPRouteRefs) > 0) {
```

Comment:
AFAIR, len would return 0 for nil obj as well. 
```suggestion
	if !(len(r.Spec.Production) > 0 && len(r.Spec.Production[0].HTTPRouteRefs) > 0) && !(len(r.Spec.Sandbox) > 0 && len(r.Spec.Sandbox[0].HTTPRouteRefs) > 0) {
```

Reply:
Fixed